FirebaseInstallations/Source/Library/FIRInstallations.m

```diff
@@ -117,12 +118,38 @@ - (instancetype)initWithAppOptions:(FIROptions *)appOptions
   return self;
 }
 
++ (void)validateAppOptions:(FIROptions *)appOptions appName:(NSString *)appName {
+  NSMutableArray *missingFields = [NSMutableArray array];
+  if (appName.length < 1) {
+    [missingFields addObject:@"`FirebaseApp.name`"];
+  }
+  if (appOptions.APIKey.length < 1) {
+    [missingFields addObject:@"`FirebaseOptions.APIKey`"];
+  }
+  if (appOptions.projectID.length < 1) {
+    [missingFields addObject:@"`FirebaseOptions.projectID`"];
+  }
+  if (appOptions.googleAppID.length < 1) {
+    [missingFields addObject:@"`FirebaseOptions.googleAppID`"];
+  }
+  if (appOptions.GCMSenderID.length < 1) {
+    [missingFields addObject:@"`FirebaseOptions.googleAppID`"];
```

Comment:
This should be `GCMSenderID`

Reply:
Good catch!

Reply:
I think, I mentioned this elsewhere, but we do NOT need GCMSenderID.
We only need GCMSenderID if projectID is not set.
Thus, I would recommend to change the check to look something like this:

if (appOptions.projectID.length < 1  &&  appOptions.GCMSenderID.length < 1) {
  [...]

Reply:
Well, technically we can omit the check at all because in the current version of Firebase (6.x.x) `GCMSenderID` is required and enforced by Firebase Core, so`GCMSenderID` is always present and the `projectID` keeps being optional. But yeah, it may look a bit confusing, I'll update the check to be more explicit and add comments.

Reply:
BTW, `GCMSenderID` is also required to retrieve IID from the old storage, but I assume that we are fine to just skip IID migration if for some reasons (e.g. in the future versions) `GCMSenderID` is missing.